doc/src/learn/cryptography/sui-offline-signing.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Offline Signing
+---
+
+Sui supports offline signing, which is signing transactions using a device not connected to a Sui network, or in a wallet implemented in a different programming language without relying on the Sui key store. The steps to implement offline signing include:
+
+ 1. Serialize the data for signing.
+ 1. Sign the serialized data. Put the serialized data in a location to sign (such as the wallet of your choice, or tools in other programming languages) and to produce a signature with the corresponding public key.
+ 1. Execute the signed transaction.
+
+## Serialize data for a transfer
+
+You must serialize transaction data following [Binary Canonical Serialization](https://crates.io/crates/bcs) (BCS). It is supported in [other languages](https://github.com/zefchain/serde-reflection#language-interoperability).
+
+The following example demonstrates how to to serialize data for a transfer using the Sui CLI. This returns serialized transaction data in Base64. Submit the raw transaction to execute as `tx_bytes`.
+ 
+```shell
+sui client serialize-transfer-sui --to $ADDRESS --sui-coin-object-id $OBJECT_ID --gas-budget 1000
+
+Raw tx_bytes to execute: $TX_BYTES
+```
+
+## Sign the serialized data
+
+You can sign the data using the device and programming language you choose. Sui accepts signatures for pure Ed25519, ECDSA Secp256k1, ECDSA Secp256r1 and native multisig. To learn more about multisig, see [Sui Multi-Signature](sui-multisig.md).
+
+The signature is committed to an intent message of the transaction data. To learn how to construct an intent message, see [Sui Intent signing](sui-intent_signing.md). 
+
+Before you pass it in to the signing API, you must first hash the intent message to 32 bytes using with Blake2b. Depending on the signing scheme, Sui performs an additional hashing is internally to the signing API. For ECDSA Secp256k1 and Secp256r1, you must use SHA-2 SHA256 as the internal hash function. For pure Ed25519, you must use SHA-512. 
```

Comment:
```suggestion
Before you pass it in to the signing API, you must first hash the intent message to 32 bytes using with BLAKE2b. Depending on the signing scheme, Sui performs an additional hashing is internally to the signing API. For ECDSA secp256k1 and secp256r1, you must use SHA-2 SHA256 as the internal hash function. For pure ed25519, you must use SHA-512. 
```
"Sui performs an additional hashing is internally to the signing API": Not sure how to fix?

Reply:
Revise to: "To be compatible with existing standards and hardware secure modules (HSMs), the signing algorithms perform additional hashing internally."